hpx/config/compiler_specific.hpp

```diff
@@ -115,5 +115,11 @@
 #define HPX_CDECL
 #endif
 
+#if defined(__has_feature)
```

Comment:
How does this relate to the `HPX_HAVE_SANITIZERS` pp constant defined above?

Reply:
Good question. `HPX_HAVE_SANITIZERS` should presumably be enabled (by the user) if *any* sanitizer is enabled, whereas `HPX_HAVE_ADDRESS_SANITIZER` is only enabled for the address sanitizer. I'm not sure if we need the distinction.

Reply:
In this case, shouldn't the above be wrapped in a `#ifdef HPX_HAVE_ADDRESS_SANITIZER/#endif`, then?

Reply:
In principle, the user could set the asan flag independently of setting the CMake variable `HPX_WITH_SANITIZERS`. This will probably lead to linker errors though since the sanitizers are linked in completely dynamically. To make everything consistent, this should be wrapped in a `#ifdef HPX_HAVE_SANITIZERS/#endif`